vizro-core/docs/pages/user-guides/data.md

```diff
@@ -333,14 +336,77 @@ Parametrized data loading is compatible with [caching](#configure-cache). The ca
 
 You cannot pass [nested parameters](parameters.md#nested-parameters) to dynamic data. You can only target the top-level arguments of the data loading function, not the nested keys in a dictionary.
 
-### Filter update limitation
+### Filters
+
+By default, when a [filter](filters.md) depends on dynamic data, the available selector values update on page refresh to reflect the latest dynamic data. The mechanism for this, including caching, works exactly like other non-control components such as `vm.Graph`. However, unlike such components, a filter can depend on multiple data sources.
+
+Vizro looks at the data source for each component in the filter's `targets`. If at least one of these is a dynamic data source then the filter behaves dynamically and the filter's selector updates on page refresh. Remember that when `targets` is not explicitly specified, a filter applies to all the components on a page that use a DataFrame including `column`.
```

Comment:
I'm a little uncomfortable with the phrase "Vizro looks at..." could we revise to make it less anthropomorphic? Something like:

"If at least one of the data sources in the filter's `targets` is a dynamic data source, the filter behaves dynamically and the filter's selector updates on page refresh... "

Reply:
This is much better, thank you!